### PR TITLE
Commit IFC 4.4 File Exported Via LF Works for Sprint 1.1

### PR DESCRIPTION
**Pull request by jared-weston-seequent**
Upload files with no geometry for sprint 1.1. This includes project set up.

### CODE REVIEW

**Comment by pjanck** [comment]
Hi @jared-weston-seequent : thank you for submitting the files! As this is your first pull request, we (the project team) need to *manually* start the checking procedure (it's a GitHub *feature*). Please remain patient until we merge the first pull request. After that, the checker will run automatically, not needing our intervention.

If you wish to see why the files are failing, please consult the FAQ to see where to obtain the detailed report.

**Comment by pjanck** [comment]
As for the errors, there seems to be a problem with the checker procedure. I'll have a look. Please wait.

**Comment by pjanck** [comment]
I believe I have an idea where the problem lies: the checker does not recognize `IfcSolidStratum` entity, since in the Tunnel version of schema proposal (i.e. 4x4) this should be `IfcGeotechnicalStratum` (see [here](https://bsi-infraroom.github.io/IFC-Documentation-Tunnel/4_4_0_0/general/HTML/link/ifcgeotechnicalstratum.htm) with `PredefinedType=SOLID`.

**Comment by SergejMuhic** [comment]
Are there no concepts from our sprints implemented?

Also, although the checker is not checking compositions (not part of any previous sprint yet), you have an issue with the placement of `IfcGeoslice`. It should have a placement referencing the placement of its container. In your case `IfcSite`. All the parts `IfcSolidStratum` placements should then reference the placement of `IfcGeoslice`.

**Comment by jared-weston-seequent** [comment]
@SergejMuhic no not yet - we are looking for a home for existing 4.3 files and were instructed to submit them here along with 4.4 files when we create them. I will withdraw this PR and re-submit when we have something more appropriate 🤞 Thanks for having a look and all the feedback!

**Comment by SergejMuhic** [comment]
No need to withdraw, failing in this deployment project setting is not the absence of success. We are all just working towards a good standard so any feedback and any datasets are good. We can use all input to work on a better schema, templates and usages and documentation.

**Comment by SergejMuhic** [comment]
Have reopened according to https://github.com/bSI-InfraRoom/IFC-Tunnel-Deployment/pull/100#issuecomment-1479115265

**Comment by jared-weston-seequent** [comment]
@SergejMuhic my understanding was that not all elements required placements- in this case, we have a parent entity (the geoslice) that acts as a container with no local placement of its own. Therefore we reference the child elements (the strata) back to the site rather than the non-existent placement of the geoslice. I'd rather not add a (0, 0, 0) placement when not required to avoid file size bloat. 

Would it be better if we just submitted files that matched what is covered in the sprints? Currently I see some files which are cut down and are only covering basics, e.g project set up, coordinate systems etc, and some which are fully fledged files (e.g that of trimble). 

I'm happy to add the modifications to replace `IfcSolidStratum` with `IfcGeotechnicalStratum` and `PredefinedType=SOLID` for now... Hopefully this will at least make the checker happy! 

image

**Comment by jared-weston-seequent** [comment]
Just a quick Q about project declaration. We don't do any of this at the moment. Is implementing this enforced in the schema / checker? I presume for our files, we would be able to define an `IfcActor` & assign the base site to this. At the moment we have no concept of 'who owns what' within a Leapfrog project, so I'm not sure we can do much more. Some things that would be nice would be to use the grouping relation to relate derived objects, e.g a model build from a set of boreholes. We also can access some coordinate system information on the example project so we can attempt the examples from sprint 2

**Comment by SergejMuhic** [comment]
> @SergejMuhic my understanding was that not all elements required placements- in this case, we have a parent entity (the geoslice) that acts as a container with no local placement of its own.

Not all elements requiring placements is correct but in this context, the following template applies:
https://bsi-infraroom.github.io/IFC-Documentation-Tunnel/4_4_0_0/general/HTML/link/element-decomposition.htm
Specifically this part:
image

The children all reference the placement of the composite and only the composite is contained (containment template) in a spatial structure element.



> I'd rather not add a (0, 0, 0) placement when not required to avoid file size bloat.

The placement should only be (0,0,0) if the composite (the whole, your IfcGeoslice) is also placed at the origin of the site. It should be a local placement related to the site placement.


> Would it be better if we just submitted files that matched what is covered in the sprints?

That was the general idea. This way we work through the templates one by one and the discussion above might resolve itself. 😄 

The checker only check the sprints until now, so it will not check any Element Decompositions, just Spatial Decompositions.

**Comment by SergejMuhic** [comment]
> Just a quick Q about project declaration. We don't do any of this at the moment. Is implementing this enforced in the schema / checker? I presume for our files, we would be able to define an `IfcActor` & assign the base site to this. At the moment we have no concept of 'who owns what' within a Leapfrog project, so I'm not sure we can do much more. Some things that would be nice would be to use the grouping relation to relate derived objects, e.g a model build from a set of boreholes. We also can access some coordinate system information on the example project so we can attempt the examples from sprint 2

Regarding declaration, please read:
https://bsi-infraroom.github.io/IFC-Documentation-Tunnel/4_4_0_0/general/HTML/link/project-declaration.htm
and
https://bsi-infraroom.github.io/IFC-Documentation-Tunnel/4_4_0_0/general/HTML/link/ifcproject.htm 
(scroll down or search for Project Declaration on the page)

Project as your root node of an IFC dataset requires all concepts to be attached to it. Apart from a few special cases such as classification and spatial decomposition, all concepts such as property set templates, libraries, types, groups or system etc. are supposed to be attached through declaration. If your software does not deal with IfcActor, I would suggest either leaving it empty or declaring one of the other possibilities that your software does support. Just add a note to your PR so that reviewers know why you modified a particular usage / concept. This way we can merge overriding the checker settings.

**Comment by jared-weston-seequent** [comment]
I have made a much more cut-down version of this project for sprint 1.1. Sorry - a little behind as getting time to work on this is difficult, but this is probably easiest. The actual alignment work is not something LF supports so we will catch up with the 1.X sprints in the mean time.

I have added a dummy property set template and declared these on the project. We will eventually use this for defining borehole set properties, so I believe this is the most relevant way for us to progress here.

File: files/Seequent/LeapfrogWorksIFC_4_4_Sprint_1_1.ifc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+DATA;
+#1=IFCPERSON($,'Weston','Jared',$,$,$,$,$);
+#2=IFCORGANIZATION($,'Seequent',$,$,$);
+#3=IFCORGANIZATION($,'Seequent',$,$,$);
+#4=IFCPERSONANDORGANIZATION(#1,#2,$);
+#5=IFCAPPLICATION(#3,'2022.1.1.dev','Leapfrog Works','Works');
+#6=IFCOWNERHISTORY(#4,#5,$,.ADDED.,1681143227,$,$,1681143227);
+#7=IFCCARTESIANPOINT((0.,0.,0.));
+#8=IFCAXIS2PLACEMENT3D(#7,$,$);
+#9=IFCGEOMETRICREPRESENTATIONCONTEXT($,'Model',3,1.0E-5,#8,$);
```
Comment (larswik) [inline files/Seequent/LeapfrogWorksIFC_4_4_Sprint_1_1.ifc:16]:
```suggestion
#9=IFCGEOMETRICREPRESENTATIONCONTEXT('3D','Model',3,1.0E-5,#8,$);
```

**Review by larswik — CHANGES_REQUESTED** [review]

**Comment by pjanck** [comment]
After the proposed fix by @larswik has been incorporated, we can merge this PR.

> I have added a dummy property set template and declared these on the project. We will eventually use this for defining borehole set properties, so I believe this is the most relevant way for us to progress here.

Great approach! There is a `WHERE` rule failing currently on the template, so make sure to fix that in the meantime as well.

**Review by pjanck — APPROVED** [review]
The file looks fine for sprint 1.1 - except for what @larswik has already noted.

**Review by larswik — APPROVED** [review]
lftm

**Pull request merged**